src/fsharp/vs/service.fs

```diff
@@ -1801,13 +1806,29 @@ type FSharpCheckProjectResults(projectFileName:string, keepAssemblyContents, err
         let (tcGlobals, tcImports, thisCcu, ccuSig, _tcSymbolUses, topAttribs, _tcAssemblyData, _ilAssemRef, _ad, _tcAssemblyExpr, _dependencyFiles) = getDetails()
         FSharpAssemblySignature(tcGlobals, thisCcu, tcImports, topAttribs, ccuSig)
 
-    member info.AssemblyContents =  
+    member info.TypedImplementionFiles =
         if not keepAssemblyContents then invalidOp "The 'keepAssemblyContents' flag must be set to true on the FSharpChecker in order to access the checked contents of assemblies"
         let (tcGlobals, tcImports, thisCcu, _ccuSig, _tcSymbolUses, _topAttribs, _tcAssemblyData, _ilAssemRef, _ad, tcAssemblyExpr, _dependencyFiles) = getDetails()
         let mimpls = 
             match tcAssemblyExpr with 
             | None -> []
             | Some mimpls -> mimpls
+        tcGlobals, thisCcu, tcImports, mimpls
+
+    member info.AssemblyContents = FSharpAssemblyContents(info.TypedImplementionFiles)
+
+    member info.OptimizedAssemblyContents =  
```

Comment:
Please change to ``GetOptimizedAssemblyContents()`` since it involves potentially very long computation (e.g. we don't want it triggering in the debugger when looking at one of these objects)